client/history/client.go

```diff
@@ -1079,6 +1080,21 @@ func (c *clientImpl) GetFailoverInfo(
 	return c.client.GetFailoverInfo(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
 }
 
+func (c *clientImpl) RatelimitUpdate(ctx context.Context, request *types.RatelimitUpdateRequest, opts ...yarpc.CallOption) (*types.RatelimitUpdateResponse, error) {
+	if len(opts) == 0 {
+		// unfortunately there is not really any way to ensure "must have a shard key option"
+		// due to the closed nature of yarpc.CallOption's implementation, outside private-field-reading reflection.
+		//
+		// we could achieve this by making this client a different interface,
+		// but that'd complicate wrappers quite a bit more.
+		return nil, fmt.Errorf("invalid arguments, missing yarpc.WithShardKey(peer) at a minimum")
+	}
```

Comment:
The arguments here are _essentially_ required to be exactly this, due to sharing this interface with the generated interface, and the various interface-wrapping generated code we've added on top of it.

I'd prefer a dedicated string / HostInfo arg to make it unambiguous, but that'd require a LOT of changes because we'd have to stop using the generated client interface basically anywhere.